web/src/components/toolbar/ColorerOption.vue

```diff
@@ -1,44 +1,54 @@
-<script>
-export default {
-  props: {
-    title: {
-      type: String,
-      required: false,
-      default: '',
-    },
-    value: {
-      type: String,
-      required: false,
-      default: null,
-    },
-    disabled: {
-      type: Boolean,
-      required: false,
-    },
-    options: { // {name: string, options: {name: string, color?: string}[]}[]
-      type: Array,
-      required: true,
-    },
-  },
-  computed: {
-    showSelect() {
-      return !this.value || this.options.length > 1;
-    },
-    filterOptions() {
-      const selected = this.options.find(d => d.value === this.value);
-      return selected ? selected.options : [];
-    },
-    hasOptions() {
-      if (this.options.length === 0) {
-        return false;
-      }
-      if (this.options.length === 1 && !this.options[0].value) {
-        return false;
-      }
-      return true;
-    },
-  },
-};
+<script lang="ts">
+import { Component, Vue, Prop } from 'vue-property-decorator';
+
+export interface IOptionGroup {
+  name: string;
+  value: string;
+  options: {name: string, color?: string}[];
+}
+
+@Component
+export default class ColorerOption extends Vue {
+  @Prop({
+    default: '',
+  })
+  readonly title!: string;
+
+  @Prop({
+    default: null,
+    type: String,
+  })
+  readonly value!: string|null;
```

Comment:
Does the `!` tell TSC that this value will not be null or undefined? If so, how does that square with the specified type (and default value)?

Reply:
generally yes, in this case it will tell typescript that the property will be initialized by someone

Reply:
usually I would define a default value. However, it is not allowed to initialize a property in that way